### PR TITLE
Feature/create table

### PR DESCRIPTION
**Pull request by per-wat**
- Created a table with dummy data
- Used MUI Table Component
- Basic styling with theme

### CODE REVIEW

File: src/components/Table/table.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableHead,
+	TableRow,
+	Paper,
+	withStyles,
+	makeStyles,
+} from "@material-ui/core"
+
```
Comment (jbyomuhangi) [inline src/components/Table/table.js]: Imports should always appear at the top of the file

File: src/components/Table/table.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
+const data = [
+	{
+		id: "cGVvcGxlOjE=",
+		name: "Luke Skywalker",
+		eyeColor: "blue",
```
Comment (jbyomuhangi) [inline src/components/Table/table.js]: Consider moving the dummy data into a separate file and importing it

**Review by jbyomuhangi — CHANGES_REQUESTED** [review]
This is a good start, one important thing I need to point out is modularity. When we make react components, we must make them in a way such that they can be reused. In this example we are making a table component, ask yourself "am I able to reuse this component if I need to?".

If we wanted to make another page the also uses a table that is very similar to this, how would you achieve this? At the moment you would have to write code that will be almost identical to this, which isn't good practice. Try to make this table component as reusable as possible

**Comment by per-wat** [comment]
The recently pushed codes are resolving the previous review

File: src/components/CustomTable/CustomisedTableHead/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import CustomisedTableHead from "./CustomisedTableHead"
+
+export default CustomisedTableHead
```
Comment (jbyomuhangi) [inline src/components/CustomTable/CustomisedTableHead/index.js]: You can simplify this to 

```
export { default } from "./CustomisedTableHead";

```

The same can be done in your other index.js filess for your components

File: src/components/CustomTable/StyledTableCell/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import StyledTableCell from "./StyledTableCell"
+
+export default StyledTableCell
```
Comment (jbyomuhangi) [inline src/components/CustomTable/StyledTableCell/index.js]: Same here

File: src/components/CustomTable/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import CustomTable from "./CustomTable"
+
+export default CustomTable
```
Comment (jbyomuhangi) [inline src/components/CustomTable/index.js]: Here too

File: src/components/CustomTable/StyledTableBody/index.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import StyledTableBody from "./StyledTableBody"
+
+export default StyledTableBody
```
Comment (jbyomuhangi) [inline src/components/CustomTable/StyledTableBody/index.js]: Here too

File: src/components/CustomTable/StyledTableBody/StyledTableBody.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    "&:nth-of-type(even)": {
+      backgroundColor: theme.palette.secondary.main,
+    },
+    "&$hover:hover": {
+      backgroundColor: theme.palette.primary.main,
+      cursor: "pointer",
+    },
+  },
+  //Produce warning if removed
+  hover: {},
```
Comment (jbyomuhangi) [inline src/components/CustomTable/StyledTableBody/StyledTableBody.js]: What warning is this giving?

File: src/components/CustomTable/StyledTableCell/StyledTableCell.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  body: {
+    borderBottomColor: theme.palette.primary.dark,
+    textTransform: "capitalize",
+  },
+}))(TableCell)
+
+const StyledTableCell = ({ children }) => {
+  return <NewTableCell>{replaceIfNull(children)}</NewTableCell>
+}
```
Comment (jbyomuhangi) [inline src/components/CustomTable/StyledTableCell/StyledTableCell.js]: I wouldn't actually recommend doing this. `children` isn't limited to a string, it's actually a react node. You should be calling `replaceIfNull` at the point you get the string.

**Review by jbyomuhangi — CHANGES_REQUESTED** [review]

File: src/components/CustomTable/StyledTableBody/StyledTableBody.js (hunk 7 above)
Comment (per-wat) [reply src/components/CustomTable/StyledTableBody/StyledTableBody.js]: It produces this warning below.

`Warning: [JSS] Could not find the referenced rule "hover" in "makeStyles".`

This also affects the styling under the hover state
```
root: {
    "&$hover:hover": {
      backgroundColor: theme.palette.primary.main,
      cursor: "pointer",
    },
  },
hover: {},
``` 

It seems that to apply the style on the hover state on that component, it is required for us to do code like that to override the default class
```
.MuiTableRow-root.MuiTableRow-hover:hover {
    background-color: rgba(0, 0, 0, 0.04);
}
```

Comment (jbyomuhangi) [reply src/components/CustomTable/StyledTableBody/StyledTableBody.js]: You can change your CSS to look like this:

image


And remove the `hover` prop from the `TableRow` component

**Review by jbyomuhangi — CHANGES_REQUESTED** [review]
Some additional notes:

- This error is still showing up in the console, please resolve it:
image

- A note on your design choice, please make the table take up the available space so the page and change the background color of the page to white as well.:
image

**Review by jbyomuhangi — APPROVED** [review]
Good job

**Pull request merged**